sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/AbstractAggregator.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics.aggregator;
+
+public abstract class AbstractAggregator implements Aggregator {
+
+  @Override
+  public void mergeToAndReset(Aggregator other) {
+    if (!other.getClass().isAssignableFrom(this.getClass())) {
```

Comment:
I heard once that getClass is expensive, maybe we do cache that in a constant for this.

Reply:
I am probably wrong, we need benchmarks :) later maybe

Reply:
heh. yeah, I think getClass() is very fast, but benchmarks can show the truth.

Reply:
also, I had this logic backwards. fixed now.